### PR TITLE
Support pulling multiple dbs in Lando

### PR DESCRIPTION
**Pull request by grasmash**
**Motivation**

Allows multiple dbs to be pulled.

**Proposed changes**

Add the `--multiple-dbs` option to the `acli pull:db` command.

**Alternatives considered**

**Testing steps**

1. Follow the [contribution guide](https://github.com/acquia/cli/blob/master/CONTRIBUTING.md#building-and-testing) to set up your development environment.
2. Clear the kernel cache to pick up new and changed commands: `./bin/acli ckc`
3. Run `acli pull:db --multiple-dbs`
4. Run `acli pull:db`
5. Test in a Cloud IDE, in Lando, and on your native local machine.

**Merge requirements**
- [x] _Bug_, _enhancement_, or _breaking change_ label applied
- [ ] Manual testing by a reviewer

### CODE REVIEW

**Comment by grasmash** [comment]
image

**Comment by grasmash** [comment]
@mikemadison13 Could you please test this?

**Review by danepowell — APPROVED** [review]

**Review by anavarre — CHANGES_REQUESTED** [review]
> Test in a Cloud IDE and locally. Ideally Lando.

I was surprised to see this because in Cloud IDE we do have 2 DBs, but one (`drupal`) is the default DB and the other (`drupal2`) is really only made for AMA. That being said, I wanted to test the experience there and surprisingly I didn't get any issue so I suspect the first DB pulled got overwritten by the other, which is **not expected**.

💡 Shouldn't we exclude Cloud IDE entirely from being able to use this parameter? 💡

```
test:~/project (master) $ /tmp/aclidev pull:db --help
Description:
  Import database backup from a Cloud Platform environment

Usage:
  pull:database [options] [--] [<environmentId> [<site>]]
  pull:db
  pull:database [<environmentAlias>]
  pull:database myapp.dev
  pull:database 12345-abcd1234-1111-2222-3333-0e02b2c3d470
  pull:database myapp.dev default

Arguments:
  environmentId         The Cloud Platform environment ID or alias
  site                  For a multisite application, the directory name of the site

Options:
      --no-scripts      Do not run any additional scripts after the database is pulled. E.g., drush cache-rebuild, drush sql-sanitize, etc.
      --on-demand       Force creation of an on-demand backup. This takes much longer than using an existing backup (when one is available)
      --no-import       Download the backup but do not import it (implies --no-scripts)
      --multiple-dbs    Download multiple dbs. Defaults to FALSE.
  -h, --help            Display help for the given command. When no command is given display help for the list command
  -q, --quiet           Do not output any message
  -V, --version         Display this application version
      --ansi|--no-ansi  Force (or disable --no-ansi) ANSI output
  -n, --no-interaction  Do not ask any interactive question
  -v|vv|vvv, --verbose  Increase the verbosity of messages: 1 for normal output, 2 for more verbose output and 3 for debug

Help:
  This uses the latest available database backup, which may be up to 24 hours old. If no backup exists, one will be created.
test:~/project (master) $ /tmp/aclidev pull:db --multiple-dbs
Using Cloud Application drupal8

 Choose a Cloud Platform environment [Dev, dev (vcs: pipelines-build-master)]:
  [0] Dev, dev (vcs: pipelines-build-master)
  [1] Prod, prod (vcs: tags/2020-09-28)
  [2] Stage, test (vcs: tags/2020-10-19)
 > 0

 ! [NOTE] This is a multisite application. Drupal will load the default site unless you've configured sites.php for this
 !        environment: https://docs.acquia.com/cloud-platform/develop/drupal/multisite/                                 

 Choose a site [default]:
  [0] drupal9
  [1] default
 > 1

 Choose databases. You may choose multiple. Use commas to separate choices. [drupal8 (default)]:
  [all] All
  [0  ] drupal8 (default)
  [1  ] drupal9
 > all

                                                                                                                        
 [INFO] Using a database backup that is 3 hours old. Backup #191076535 was created at Wed Nov 10 7:12:09 UTC 2021.      
                                                                                                                        
        You can view your backups here:                                                                                 
        https://cloud.acquia.com/a/environments/18672-0e737068-176c-2b7c-8b6e-3hdf0561b428/databases                    
                                                                                                                        
        To generate a new backup, re-run this command with the --on-demand option.                                      
                                                                                                                        

    ✔ Downloading drupal8 database copy from the Cloud Platform
    ⌛ Importing drupal8 database download...

    ✔ Importing drupal8 database download
                                                                                                                        
 [INFO] Using a database backup that is 3 hours old. Backup #191076536 was created at Wed Nov 10 7:12:10 UTC 2021.      
                                                                                                                        
        You can view your backups here:                                                                                 
        https://cloud.acquia.com/a/environments/18672-0e737068-176c-2b7c-8b6e-3hdf0561b428/databases                    
                                                                                                                        
        To generate a new backup, re-run this command with the --on-demand option.                                      
                                                                                                                        

    ✔ Downloading drupal9 database copy from the Cloud Platform
    ⌛ Importing drupal9 database download...
    ✔ Importing drupal9 database download
        }
test:~/project (master) $ mysql -u drupal -pdrupal
mysql: [Warning] Using a password on the command line interface can be insecure.
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 10
Server version: 5.7.34 MySQL Community Server (GPL)

Copyright (c) 2000, 2021, Oracle and/or its affiliates.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> show databases;
+--------------------+
| Database           |
+--------------------+
| information_schema |
| drupal             |
| drupal2            |
+--------------------+
3 rows in set (0.00 sec)

mysql> use drupal2;
Database changed
mysql> show tables;
Empty set (0.00 sec)

mysql> 
```

**Comment by grasmash** [comment]
Added intentional failure in Cloud IDE: 
image

File: src/Command/Pull/PullCommandBase.php [hunk 1]
```diff
@@ -569,6 +576,9 @@ protected function promptChooseDatabases(
       $chosen_database_keys = $this->io->askQuestion($question);
       $chosen_databases = [];
       if (count($chosen_database_keys) === 1 && $chosen_database_keys[0] === 'all') {
+        if (count($environment_databases) > 10) {
+          $this->io->warning('You have chosen to pull down more than 10 databases. This could exhaust your disk space.');
```
Comment (anavarre) [inline src/Command/Pull/PullCommandBase.php:580]: We cannot do this in Cloud IDE until we're checking for the available disk space prior.

File: src/Command/Pull/PullCommandBase.php [hunk 2]
```diff
@@ -356,7 +374,13 @@ protected function connectToLocalDatabase($db_host, $db_user, $db_name, $db_pass
     ];
     $process = $this->localMachineHelper->execute($command, $output_callback, NULL, FALSE, NULL, ['MYSQL_PWD' => $db_password]);
     if (!$process->isSuccessful()) {
-      throw new AcquiaCliException('Unable to connect to local database. {message}', ['message' => $process->getErrorOutput()]);
+      throw new AcquiaCliException('Unable to connect to local database using credentials mysql:://{user}:{password}@{host}/{database}. {message}', [
+        'message' => $process->getErrorOutput(),
+        'host' => $db_host,
+        'user' => $db_user,
+        'password' => $db_password,
```
Comment (grasmash) [inline src/Command/Pull/PullCommandBase.php:381]: Is this ok?

**Comment by anavarre** [comment]
Can you please take the opportunity of this PR to replace "in" by "it" in https://github.com/acquia/cli/blob/d2733ccc3ef0a2b42b1bce3930b51bd0d1e62c8f/src/Helpers/LocalMachineHelper.php#L73 ?

**Comment by mikemadison13** [comment]
i think i have a few questions here:

- what happens if the databases in the clouds do not directly match up with the names of sites in my local config? theoretically, since the database includes can be handled manually in settings.php i can name something local foo and my database bar and it "should" work on cloud classic (i've managed to do this on accident, and it's causing the ACN team all measure of headaches) 
- what happens if we are trying to do some of the new hook integration with the composer scripts that was added in https://github.com/acquia/cli/pull/648 ? if i pull 4 databases, and i want to automate database updates and config imports for each, is that possible? otherwise i think what you've outlined here looks good and i don't want to hold you up.

**Comment by mikemadison13** [comment]
FYI @grasmash I pulled the code from this PR, compiled the phar, and am attempting to test with `pull:db --multiple-dbs` arguments and it's not giving me the choice of "all"

```
./acli.phar pull:db --multiple-dbs
Using Cloud Application Mike Madison (Employee)

 Choose a Cloud Platform environment [Dev, dev (vcs: pipelines-build-main)]:
  [0] Dev, dev (vcs: pipelines-build-main)
  [1] Next, next (vcs: tags/WELCOME)
  [2] Prod, prod (vcs: tags/WELCOME)
  [3] Stage, test (vcs: tags/WELCOME)
 > 0

 ! [NOTE] This is a multisite application. Drupal will load the default site unless you've configured sites.php for this
 !        environment: https://docs.acquia.com/cloud-platform/develop/drupal/multisite/                                 

 Choose a site [default]:
  [0] default
  [1] multisite
  [2] second
  [3] third

```

**Comment by grasmash** [comment]
Please share the verbose output

**Comment by mikemadison13** [comment]
```
./acli.phar pull:db --multiple-dbs -vvv

Box Requirements Checker
========================

> Using PHP 8.0.3
> PHP is using the following php.ini file:
  WARNING: No configuration file (php.ini) used by PHP!

> Checking Box requirements:
  ✔ The application requires the version "^7.3 | ^8.0" or greater.
  ✔ The application requires the extension "json".
  ✔ The package "brick/math" requires the extension "json".
  ✔ The package "guzzlehttp/guzzle" requires the extension "json".
  ✔ The package "ramsey/uuid" requires the extension "json".
  ✔ The package "zumba/amplitude-php" requires the extension "curl".
  
                                                                                                                                                                                          
 [OK] Your system is ready to run the application.                                                                                                                                        
                                                                                                                                                                                          

Composer script `pre-acli-pull-database` does not exist in `/app/composer.json`, skipping.
Acquia CLI version: 1.18.1-23-g5c46db7
Connecting to database drupal9
[notice] Command: 'which' 'mysql' [Exit: 0]
[notice] Command: 'mysql' '--host' 'database.multisite.internal' '--user' 'drupal9' 'drupal9' [Exit: 0]
[debug] Using Cloud application UUID b025f101-d6bf-413f-8483-33072a85146d from blt/blt.yml
Using Cloud Application Mike Madison (Employee)

 Choose a Cloud Platform environment [Dev, dev (vcs: pipelines-build-main)]:
  [0] Dev, dev (vcs: pipelines-build-main)
  [1] Next, next (vcs: tags/WELCOME)
  [2] Prod, prod (vcs: tags/WELCOME)
  [3] Stage, test (vcs: tags/WELCOME)
 > 0

[debug] Using environment Dev 97227-b025f101-d6bf-413f-8483-33072a85146d
[notice] Command: 'which' 'ssh' [Exit: 0]
[notice] Command: 'ssh' 'eemmadison.dev@eemmadisondev.ssh.prod.acquia-sites.com' '-t' '-o StrictHostKeyChecking=no' '-o AddressFamily inet' '-o LogLevel=ERROR' 'ls' '/mnt/files/eemmadison.dev/sites' [Exit: 0]
[debug] Command:  ls /mnt/files/eemmadison.dev/sites [Exit: 0]
[debug] Multisite detected
 ! [NOTE] This is a multisite application. Drupal will load the default site unless you've configured sites.php for this
 !        environment: https://docs.acquia.com/cloud-platform/develop/drupal/multisite/                                 

 Choose a site [default]:
  [0] default
  [1] multisite
  [2] second
  [3] third

```

**Comment by grasmash** [comment]
I think the way it's written now it will work for multiple dbs for a single site but not multisite. I think we just need a small change to support multisite. I'll take a look.

**Comment by grasmash** [comment]
I'll merge this as is and follow up with multisite support. The current PR adds single site, multi db support.

**Pull request merged**